cobalt/browser/application.cc

```diff
@@ -212,9 +212,18 @@ GURL GetInitialURL(bool should_preload) {
   GURL initial_url = GURL(kDefaultURL);
   // Allow the user to override the default URL via a command line parameter.
   base::CommandLine* command_line = base::CommandLine::ForCurrentProcess();
-  if (command_line->HasSwitch(switches::kInitialURL)) {
+  bool has_initial_url_switch = command_line->HasSwitch(switches::kInitialURL);
+  const auto& argv = command_line->argv();
+  bool last_argv_is_not_a_switch =
+      (argv.size() > 1) && argv.back().substr(0, 2) != "--";
+  LOG(INFO) << __FUNCTION__
+            << " last_argv_is_not_a_switch=" << last_argv_is_not_a_switch
+            << " argv.size()=" << argv.size() << " argv.back()=" << argv.back();
+  if (has_initial_url_switch || last_argv_is_not_a_switch) {
     std::string url_switch =
-        command_line->GetSwitchValueASCII(switches::kInitialURL);
+        has_initial_url_switch
+            ? command_line->GetSwitchValueASCII(switches::kInitialURL)
+            : argv.back();
```

Comment:
Maybe I'm wrong about this, but isn't it possible for switch values to be passed with a space separating it from the switch like `--switch value` instead of `--switch=value`? If so, I think it would give this issues

It also seems weird to me to take the last cli argument if it's not a switch in the first place instead of just taking whatever arguments don't have a flag before them—is this just to make the cli easier to use or does it have some greater purpose?

Reply:
I tried to make it behave more similar to chrome, but you're right this is probably incomplete. I'll dive into it some more.

Reply:
For the record, I looked it up.

The POSIX guidelines for command-line arguments (*1) only specifies command-line options as preceded by a single minus '-', and occurence of the double minus '--' is used as a separator to indicate the end of command-line options. It guides the use of 'getopt (*2)' to decode a command-line.

The double minus command-line option is a GNU extension (*3) called 'long option', and according to the libc manual should be formatted '--name=value' and can be decoded with 'getopt_long' (*4). The linux man page for long parameters allows both '--arg=param' and '--arg param' (*5).

Chrome `base::CommandLine` does not use 'getopt' nor 'getopt_long'. The Chromium binary allows switches after the URL as well as before the URL, and changing the ordering does not change the behavior.

I'll dig into where/how Chromium does it's parsing and see if I can mirror it more closely here. Since we also use 'base::CommandLine' it seems that it should be quite possible.

*1: https://pubs.opengroup.org/onlinepubs/9799919799/basedefs/V1_chap12.html
*2: https://pubs.opengroup.org/onlinepubs/9699919799/functions/getopt.html
*3: https://www.gnu.org/software/libc/manual/html_node/Argument-Syntax.html
*4: https://www.gnu.org/software/libc/manual/html_node/Getopt.html
*5: https://linux.die.net/man/3/getopt_long

Reply:
Changed it to just use `command_line->GetArgs()` just like Chromium does.